app/DynamicGraphs/CourseFinder.hs

```diff
@@ -0,0 +1,41 @@
+{-|
+    Module      : DynamicGraphs.CourseFinder
+    Description : Retrieve course information needed to generate graphs.
+
+This module contains the logic requests for information needed to
+generate graphs before retrieving and structuring the necessary data.
+-}
+module DynamicGraphs.CourseFinder (lookupCourses) where
+
+import qualified Data.Text.Lazy as T
+import qualified Data.Map.Strict as Map
+import Control.Monad.Trans.State
```

Comment:
It would be good to explicitly list what you're importing here (`StateT`, `modify`, etc.?)